### ISSUE TITLE
Placement new and auto

### ISSUE THREAD

**Issue by flamewing** [microsoft/vscode-cpptools#6246]
**Type: LanguageService**

**Describe the bug**
- OS and Version: Happens on Windows and Ubuntu
- VS Code Version: 1.49.2
- C/C++ Extension Version: 1.0.1
- Other extensions you installed (and if the issue persists after disabling them): happens with C/C++ extension on its own
- Does this issue involve using SSH remote to run the extension on a remote machine?: No
- Intellisense fails to deduce return type of placement new when assigning the result to an auto variable.

**Steps to reproduce**
1. Enable the C/C++ extension
2. Create a c++ code file with the following code:
```c++
#include <new>

struct Test { /* stuff */ };

void f() {
    alignas(alignof(Test)) unsigned char buffer[sizeof(Test)];
    auto *pf = new (buffer) Test;
    // Use pf
    pf->~Test();
}
```
3. Save and let Intellisense run
4. Intellisense highlights the `auto` keyword as an error, and displays the following error message:
> cannot deduce 'auto' type C/C++ (1587) [7, 5]

Additionally, it displays an error that it cannot find the correct overload to use; this may be related to issue #5793 .

**Expected behavior**
This should not be flagged as an issue because the type is [well specified](https://godbolt.org/z/bM3ern).

Extension logs (though they seem to be uninteresting):

<details>
  <summary><strong>Logs</strong></summary>

```
cpptools/didChangeCppProperties
Attempting to get defaults from compiler in "compilerPath" property: 'C:/msys64/mingw64/bin/g++.exe'
Compiler probe command line: C:/msys64/mingw64/bin/g++.exe -std=c11 -Wp,-v -E -dD -x c -m64 nul
Code browsing service initialized
Attempting to get defaults from compiler in "compilerPath" property: 'C:/msys64/mingw64/bin/g++.exe'
Compiler probe command line: C:/msys64/mingw64/bin/g++.exe -std=c++17 -Wp,-v -E -dD -x c++ -m64 nul
  Folder: C:/MSYS64/MINGW64/LIB/GCC/X86_64-W64-MINGW32/10.2.0/INCLUDE/ will be indexed
  Folder: C:/MSYS64/MINGW64/INCLUDE/ will be indexed
  Folder: C:/MSYS64/MINGW64/LIB/GCC/X86_64-W64-MINGW32/10.2.0/INCLUDE-FIXED/ will be indexed
  Folder: C:/MSYS64/MINGW64/X86_64-W64-MINGW32/INCLUDE/ will be indexed
  Folder: C:/MSYS64/HOME/FLAMEWING/TEMP/ will be indexed
  Folder: C:/MSYS64/MINGW64/LIB/GLIB-2.0/INCLUDE/ will be indexed
  Folder: C:/MSYS64/MINGW64/LIB/GTK-2.0/INCLUDE/ will be indexed
Populate include completion cache.
textDocument/didOpen: file:///c%3A/msys64/home/flamewing/temp/test.cc
Discovering files...
cpptools/activeDocumentChange: file:///c%3A/msys64/home/flamewing/temp/test.cc
cpptools/textEditorSelectionChange
cpptools/getDocumentSymbols: file:///c%3A/msys64/home/flamewing/temp/test.cc (id: 2)
cpptools/getFoldingRanges: file:///c%3A/msys64/home/flamewing/temp/test.cc (id: 3)
cpptools/getSemanticTokens: file:///c%3A/msys64/home/flamewing/temp/test.cc (id: 4)
cpptools/getDocumentSymbols: file:///c%3A/msys64/home/flamewing/temp/test.cc (id: 5)
cpptools/getDocumentSymbols: file:///c%3A/msys64/home/flamewing/temp/test.cc (id: 6)
cpptools/abortRequest
cpptools/getFoldingRanges: file:///c%3A/msys64/home/flamewing/temp/test.cc (id: 7)
cpptools/getDocumentSymbols
cpptools/getDocumentSymbols
cpptools/getDocumentSymbols
idle loop: reparsing the active document
Checking for syntax errors: file:///c%3A/msys64/home/flamewing/temp/test.cc
  Processing folder (recursive): C:/MSYS64/MINGW64/LIB/GCC/X86_64-W64-MINGW32/10.2.0/INCLUDE/
  Processing folder (recursive): C:/MSYS64/MINGW64/INCLUDE/
sending compilation args for C:\MSYS64\HOME\FLAMEWING\TEMP\TEST.CC
  include: C:\MSYS64\MINGW64\LIB\GLIB-2.0\INCLUDE
  include: C:\MSYS64\MINGW64\LIB\GTK-2.0\INCLUDE
  include: C:\MSYS64\MINGW64\INCLUDE\C++\10.2.0
  include: C:\MSYS64\MINGW64\INCLUDE\C++\10.2.0\X86_64-W64-MINGW32
  include: C:\MSYS64\MINGW64\INCLUDE\C++\10.2.0\BACKWARD
  include: C:\MSYS64\MINGW64\LIB\GCC\X86_64-W64-MINGW32\10.2.0\INCLUDE
  include: C:\MSYS64\MINGW64\INCLUDE
  include: C:\MSYS64\MINGW64\LIB\GCC\X86_64-W64-MINGW32\10.2.0\INCLUDE-FIXED
  include: C:\MSYS64\MINGW64\X86_64-W64-MINGW32\INCLUDE
  define: __STDC__=1
  define: __cplusplus=201703L
  define: __STDC_UTF_16__=1
  define: __STDC_UTF_32__=1
  define: __STDC_HOSTED__=1
  define: __GNUC__=10
  define: __GNUC_MINOR__=2
  define: __GNUC_PATCHLEVEL__=0
  define: __VERSION__="10.2.0"
  define: __ATOMIC_RELAXED=0
  define: __ATOMIC_SEQ_CST=5
  define: __ATOMIC_ACQUIRE=2
  define: __ATOMIC_RELEASE=3
  define: __ATOMIC_ACQ_REL=4
  define: __ATOMIC_CONSUME=1
  define: __pic__=1
  define: __PIC__=1
  define: __FINITE_MATH_ONLY__=0
  define: __SIZEOF_INT__=4
  define: __SIZEOF_LONG__=4
  define: __SIZEOF_LONG_LONG__=8
  define: __SIZEOF_SHORT__=2
  define: __SIZEOF_FLOAT__=4
  define: __SIZEOF_DOUBLE__=8
  define: __SIZEOF_LONG_DOUBLE__=16
  define: __SIZEOF_SIZE_T__=8
  define: __CHAR_BIT__=8
  define: __BIGGEST_ALIGNMENT__=16
  define: __ORDER_LITTLE_ENDIAN__=1234
  define: __ORDER_BIG_ENDIAN__=4321
  define: __ORDER_PDP_ENDIAN__=3412
  define: __BYTE_ORDER__=__ORDER_LITTLE_ENDIAN__
  define: __FLOAT_WORD_ORDER__=__ORDER_LITTLE_ENDIAN__
  define: __SIZEOF_POINTER__=8
  define: __GNUG__=10
  define: __SIZE_TYPE__=long long unsigned int
  define: __PTRDIFF_TYPE__=long long int
  define: __WCHAR_TYPE__=short unsigned int
  define: __WINT_TYPE__=short unsigned int
  define: __INTMAX_TYPE__=long long int
  define: __UINTMAX_TYPE__=long long unsigned int
  define: __CHAR16_TYPE__=short unsigned int
  define: __CHAR32_TYPE__=unsigned int
  define: __SIG_ATOMIC_TYPE__=int
  define: __INT8_TYPE__=signed char
  define: __INT16_TYPE__=short int
  define: __INT32_TYPE__=int
  define: __INT64_TYPE__=long long int
  define: __UINT8_TYPE__=unsigned char
  define: __UINT16_TYPE__=short unsigned int
  define: __UINT32_TYPE__=unsigned int
  define: __UINT64_TYPE__=long long unsigned int
  define: __INT_LEAST8_TYPE__=signed char
  define: __INT_LEAST16_TYPE__=short int
  define: __INT_LEAST32_TYPE__=int
  define: __INT_LEAST64_TYPE__=long long int
  define: __UINT_LEAST8_TYPE__=unsigned char
  define: __UINT_LEAST16_TYPE__=short unsigned int
  define: __UINT_LEAST32_TYPE__=unsigned int
  define: __UINT_LEAST64_TYPE__=long long unsigned int
  define: __INT_FAST8_TYPE__=signed char
  define: __INT_FAST16_TYPE__=short int
  define: __INT_FAST32_TYPE__=int
  define: __INT_FAST64_TYPE__=long long int
  define: __UINT_FAST8_TYPE__=unsigned char
  define: __UINT_FAST16_TYPE__=short unsigned int
  define: __UINT_FAST32_TYPE__=unsigned int
  define: __UINT_FAST64_TYPE__=long long unsigned int
  define: __INTPTR_TYPE__=long long int
  define: __UINTPTR_TYPE__=long long unsigned int
  define: __GXX_WEAK__=1
  define: __DEPRECATED=1
  define: __GXX_RTTI=1
  define: __GXX_EXPERIMENTAL_CXX0X__=1
  define: __STDCPP_DEFAULT_NEW_ALIGNMENT__=16
  define: __EXCEPTIONS=1
  define: __GXX_ABI_VERSION=1014
  define: __SCHAR_MAX__=0x7f
  define: __SHRT_MAX__=0x7fff
  define: __INT_MAX__=0x7fffffff
  define: __LONG_MAX__=0x7fffffffL
  define: __LONG_LONG_MAX__=0x7fffffffffffffffLL
  define: __WCHAR_MAX__=0xffff
  define: __WCHAR_MIN__=0
  define: __WINT_MAX__=0xffff
  define: __WINT_MIN__=0
  define: __PTRDIFF_MAX__=0x7fffffffffffffffLL
  define: __SIZE_MAX__=0xffffffffffffffffULL
  define: __SCHAR_WIDTH__=8
  define: __SHRT_WIDTH__=16
  define: __INT_WIDTH__=32
  define: __LONG_WIDTH__=32
  define: __LONG_LONG_WIDTH__=64
  define: __WCHAR_WIDTH__=16
  define: __WINT_WIDTH__=16
  define: __PTRDIFF_WIDTH__=64
  define: __SIZE_WIDTH__=64
  define: __INTMAX_MAX__=0x7fffffffffffffffLL
  define: __INTMAX_C(c)=c ## LL
  define: __UINTMAX_MAX__=0xffffffffffffffffULL
  define: __UINTMAX_C(c)=c ## ULL
  define: __INTMAX_WIDTH__=64
  define: __SIG_ATOMIC_MAX__=0x7fffffff
  define: __SIG_ATOMIC_MIN__=(-__SIG_ATOMIC_MAX__ - 1)
  define: __SIG_ATOMIC_WIDTH__=32
  define: __INT8_MAX__=0x7f
  define: __INT16_MAX__=0x7fff
  define: __INT32_MAX__=0x7fffffff
  define: __INT64_MAX__=0x7fffffffffffffffLL
  define: __UINT8_MAX__=0xff
  define: __UINT16_MAX__=0xffff
  define: __UINT32_MAX__=0xffffffffU
  define: __UINT64_MAX__=0xffffffffffffffffULL
  define: __INT_LEAST8_MAX__=0x7f
  define: __INT8_C(c)=c
  define: __INT_LEAST8_WIDTH__=8
  define: __INT_LEAST16_MAX__=0x7fff
  define: __INT16_C(c)=c
  define: __INT_LEAST16_WIDTH__=16
  define: __INT_LEAST32_MAX__=0x7fffffff
  define: __INT32_C(c)=c
  define: __INT_LEAST32_WIDTH__=32
  define: __INT_LEAST64_MAX__=0x7fffffffffffffffLL
  define: __INT64_C(c)=c ## LL
  define: __INT_LEAST64_WIDTH__=64
  define: __UINT_LEAST8_MAX__=0xff
  define: __UINT8_C(c)=c
  define: __UINT_LEAST16_MAX__=0xffff
  define: __UINT16_C(c)=c
  define: __UINT_LEAST32_MAX__=0xffffffffU
  define: __UINT32_C(c)=c ## U
  define: __UINT_LEAST64_MAX__=0xffffffffffffffffULL
  define: __UINT64_C(c)=c ## ULL
  define: __INT_FAST8_MAX__=0x7f
  define: __INT_FAST8_WIDTH__=8
  define: __INT_FAST16_MAX__=0x7fff
  define: __INT_FAST16_WIDTH__=16
  define: __INT_FAST32_MAX__=0x7fffffff
  define: __INT_FAST32_WIDTH__=32
  define: __INT_FAST64_MAX__=0x7fffffffffffffffLL
  define: __INT_FAST64_WIDTH__=64
  define: __UINT_FAST8_MAX__=0xff
  define: __UINT_FAST16_MAX__=0xffff
  define: __UINT_FAST32_MAX__=0xffffffffU
  define: __UINT_FAST64_MAX__=0xffffffffffffffffULL
  define: __INTPTR_MAX__=0x7fffffffffffffffLL
  define: __INTPTR_WIDTH__=64
  define: __UINTPTR_MAX__=0xffffffffffffffffULL
  define: __GCC_IEC_559=2
  define: __GCC_IEC_559_COMPLEX=2
  define: __FLT_EVAL_METHOD__=0
  define: __FLT_EVAL_METHOD_TS_18661_3__=0
  define: __DEC_EVAL_METHOD__=2
  define: __FLT_RADIX__=2
  define: __FLT_MANT_DIG__=24
  define: __FLT_DIG__=6
  define: __FLT_MIN_EXP__=(-125)
  define: __FLT_MIN_10_EXP__=(-37)
  define: __FLT_MAX_EXP__=128
  define: __FLT_MAX_10_EXP__=38
  define: __FLT_DECIMAL_DIG__=9
  define: __FLT_MAX__=3.40282346638528859811704183484516925e+38F
  define: __FLT_NORM_MAX__=3.40282346638528859811704183484516925e+38F
  define: __FLT_MIN__=1.17549435082228750796873653722224568e-38F
  define: __FLT_EPSILON__=1.19209289550781250000000000000000000e-7F
  define: __FLT_DENORM_MIN__=1.40129846432481707092372958328991613e-45F
  define: __FLT_HAS_DENORM__=1
  define: __FLT_HAS_INFINITY__=1
  define: __FLT_HAS_QUIET_NAN__=1
  define: __DBL_MANT_DIG__=53
  define: __DBL_DIG__=15
  define: __DBL_MIN_EXP__=(-1021)
  define: __DBL_MIN_10_EXP__=(-307)
  define: __DBL_MAX_EXP__=1024
  define: __DBL_MAX_10_EXP__=308
  define: __DBL_DECIMAL_DIG__=17
  define: __DBL_MAX__=double(1.79769313486231570814527423731704357e+308L)
  define: __DBL_NORM_MAX__=double(1.79769313486231570814527423731704357e+308L)
  define: __DBL_MIN__=double(2.22507385850720138309023271733240406e-308L)
  define: __DBL_EPSILON__=double(2.22044604925031308084726333618164062e-16L)
  define: __DBL_DENORM_MIN__=double(4.94065645841246544176568792868221372e-324L)
  define: __DBL_HAS_DENORM__=1
  define: __DBL_HAS_INFINITY__=1
  define: __DBL_HAS_QUIET_NAN__=1
  define: __LDBL_MANT_DIG__=64
  define: __LDBL_DIG__=18
  define: __LDBL_MIN_EXP__=(-16381)
  define: __LDBL_MIN_10_EXP__=(-4931)
  define: __LDBL_MAX_EXP__=16384
  define: __LDBL_MAX_10_EXP__=4932
  define: __DECIMAL_DIG__=21
  define: __LDBL_DECIMAL_DIG__=21
  define: __LDBL_MAX__=1.18973149535723176502126385303097021e+4932L
  define: __LDBL_NORM_MAX__=1.18973149535723176502126385303097021e+4932L
  define: __LDBL_MIN__=3.36210314311209350626267781732175260e-4932L
  define: __LDBL_EPSILON__=1.08420217248550443400745280086994171e-19L
  define: __LDBL_DENORM_MIN__=3.64519953188247460252840593361941982e-4951L
  define: __LDBL_HAS_DENORM__=1
  define: __LDBL_HAS_INFINITY__=1
  define: __LDBL_HAS_QUIET_NAN__=1
  define: __FLT32_MANT_DIG__=24
  define: __FLT32_DIG__=6
  define: __FLT32_MIN_EXP__=(-125)
  define: __FLT32_MIN_10_EXP__=(-37)
  define: __FLT32_MAX_EXP__=128
  define: __FLT32_MAX_10_EXP__=38
  define: __FLT32_DECIMAL_DIG__=9
  define: __FLT32_MAX__=3.40282346638528859811704183484516925e+38F32
  define: __FLT32_NORM_MAX__=3.40282346638528859811704183484516925e+38F32
  define: __FLT32_MIN__=1.17549435082228750796873653722224568e-38F32
  define: __FLT32_EPSILON__=1.19209289550781250000000000000000000e-7F32
  define: __FLT32_DENORM_MIN__=1.40129846432481707092372958328991613e-45F32
  define: __FLT32_HAS_DENORM__=1
  define: __FLT32_HAS_INFINITY__=1
  define: __FLT32_HAS_QUIET_NAN__=1
  define: __FLT64_MANT_DIG__=53
  define: __FLT64_DIG__=15
  define: __FLT64_MIN_EXP__=(-1021)
  define: __FLT64_MIN_10_EXP__=(-307)
  define: __FLT64_MAX_EXP__=1024
  define: __FLT64_MAX_10_EXP__=308
  define: __FLT64_DECIMAL_DIG__=17
  define: __FLT64_MAX__=1.79769313486231570814527423731704357e+308F64
  define: __FLT64_NORM_MAX__=1.79769313486231570814527423731704357e+308F64
  define: __FLT64_MIN__=2.22507385850720138309023271733240406e-308F64
  define: __FLT64_EPSILON__=2.22044604925031308084726333618164062e-16F64
  define: __FLT64_DENORM_MIN__=4.94065645841246544176568792868221372e-324F64
  define: __FLT64_HAS_DENORM__=1
  define: __FLT64_HAS_INFINITY__=1
  define: __FLT64_HAS_QUIET_NAN__=1
  define: __FLT128_MANT_DIG__=113
  define: __FLT128_DIG__=33
  define: __FLT128_MIN_EXP__=(-16381)
  define: __FLT128_MIN_10_EXP__=(-4931)
  define: __FLT128_MAX_EXP__=16384
  define: __FLT128_MAX_10_EXP__=4932
  define: __FLT128_DECIMAL_DIG__=36
  define: __FLT128_MAX__=1.18973149535723176508575932662800702e+4932F128
  define: __FLT128_NORM_MAX__=1.18973149535723176508575932662800702e+4932F128
  define: __FLT128_MIN__=3.36210314311209350626267781732175260e-4932F128
  define: __FLT128_EPSILON__=1.92592994438723585305597794258492732e-34F128
  define: __FLT128_DENORM_MIN__=6.47517511943802511092443895822764655e-4966F128
  define: __FLT128_HAS_DENORM__=1
  define: __FLT128_HAS_INFINITY__=1
  define: __FLT128_HAS_QUIET_NAN__=1
  define: __FLT32X_MANT_DIG__=53
  define: __FLT32X_DIG__=15
  define: __FLT32X_MIN_EXP__=(-1021)
  define: __FLT32X_MIN_10_EXP__=(-307)
  define: __FLT32X_MAX_EXP__=1024
  define: __FLT32X_MAX_10_EXP__=308
  define: __FLT32X_DECIMAL_DIG__=17
  define: __FLT32X_MAX__=1.79769313486231570814527423731704357e+308F32x
  define: __FLT32X_NORM_MAX__=1.79769313486231570814527423731704357e+308F32x
  define: __FLT32X_MIN__=2.22507385850720138309023271733240406e-308F32x
  define: __FLT32X_EPSILON__=2.22044604925031308084726333618164062e-16F32x
  define: __FLT32X_DENORM_MIN__=4.94065645841246544176568792868221372e-324F32x
  define: __FLT32X_HAS_DENORM__=1
  define: __FLT32X_HAS_INFINITY__=1
  define: __FLT32X_HAS_QUIET_NAN__=1
  define: __FLT64X_MANT_DIG__=64
  define: __FLT64X_DIG__=18
  define: __FLT64X_MIN_EXP__=(-16381)
  define: __FLT64X_MIN_10_EXP__=(-4931)
  define: __FLT64X_MAX_EXP__=16384
  define: __FLT64X_MAX_10_EXP__=4932
  define: __FLT64X_DECIMAL_DIG__=21
  define: __FLT64X_MAX__=1.18973149535723176502126385303097021e+4932F64x
  define: __FLT64X_NORM_MAX__=1.18973149535723176502126385303097021e+4932F64x
  define: __FLT64X_MIN__=3.36210314311209350626267781732175260e-4932F64x
  define: __FLT64X_EPSILON__=1.08420217248550443400745280086994171e-19F64x
  define: __FLT64X_DENORM_MIN__=3.64519953188247460252840593361941982e-4951F64x
  define: __FLT64X_HAS_DENORM__=1
  define: __FLT64X_HAS_INFINITY__=1
  define: __FLT64X_HAS_QUIET_NAN__=1
  define: __DEC32_MANT_DIG__=7
  define: __DEC32_MIN_EXP__=(-94)
  define: __DEC32_MAX_EXP__=97
  define: __DEC32_MIN__=1E-95DF
  define: __DEC32_MAX__=9.999999E96DF
  define: __DEC32_EPSILON__=1E-6DF
  define: __DEC32_SUBNORMAL_MIN__=0.000001E-95DF
  define: __DEC64_MANT_DIG__=16
  define: __DEC64_MIN_EXP__=(-382)
  define: __DEC64_MAX_EXP__=385
  define: __DEC64_MIN__=1E-383DD
  define: __DEC64_MAX__=9.999999999999999E384DD
  define: __DEC64_EPSILON__=1E-15DD
  define: __DEC64_SUBNORMAL_MIN__=0.000000000000001E-383DD
  define: __DEC128_MANT_DIG__=34
  define: __DEC128_MIN_EXP__=(-6142)
  define: __DEC128_MAX_EXP__=6145
  define: __DEC128_MIN__=1E-6143DL
  define: __DEC128_MAX__=9.999999999999999999999999999999999E6144DL
  define: __DEC128_EPSILON__=1E-33DL
  define: __DEC128_SUBNORMAL_MIN__=0.000000000000000000000000000000001E-6143DL
  define: __REGISTER_PREFIX__=
  define: __USER_LABEL_PREFIX__=
  define: __GNUC_STDC_INLINE__=1
  define: __NO_INLINE__=1
  define: __STRICT_ANSI__=1
  define: __WCHAR_UNSIGNED__=1
  define: __GCC_HAVE_SYNC_COMPARE_AND_SWAP_1=1
  define: __GCC_HAVE_SYNC_COMPARE_AND_SWAP_2=1
  define: __GCC_HAVE_SYNC_COMPARE_AND_SWAP_4=1
  define: __GCC_HAVE_SYNC_COMPARE_AND_SWAP_8=1
  define: __GCC_ATOMIC_BOOL_LOCK_FREE=2
  define: __GCC_ATOMIC_CHAR_LOCK_FREE=2
  define: __GCC_ATOMIC_CHAR16_T_LOCK_FREE=2
  define: __GCC_ATOMIC_CHAR32_T_LOCK_FREE=2
  define: __GCC_ATOMIC_WCHAR_T_LOCK_FREE=2
  define: __GCC_ATOMIC_SHORT_LOCK_FREE=2
  define: __GCC_ATOMIC_INT_LOCK_FREE=2
  define: __GCC_ATOMIC_LONG_LOCK_FREE=2
  define: __GCC_ATOMIC_LLONG_LOCK_FREE=2
  define: __GCC_ATOMIC_TEST_AND_SET_TRUEVAL=1
  define: __GCC_ATOMIC_POINTER_LOCK_FREE=2
  define: __HAVE_SPECULATION_SAFE_VALUE=1
  define: __PRAGMA_REDEFINE_EXTNAME=1
  define: __SIZEOF_INT128__=16
  define: __SIZEOF_WCHAR_T__=2
  define: __SIZEOF_WINT_T__=2
  define: __SIZEOF_PTRDIFF_T__=8
  define: __amd64=1
  define: __amd64__=1
  define: __x86_64=1
  define: __x86_64__=1
  define: __SIZEOF_FLOAT80__=16
  define: __SIZEOF_FLOAT128__=16
  define: __ATOMIC_HLE_ACQUIRE=65536
  define: __ATOMIC_HLE_RELEASE=131072
  define: __GCC_ASM_FLAG_OUTPUTS__=1
  define: __k8=1
  define: __k8__=1
  define: __code_model_medium__=1
  define: __MMX__=1
  define: __SSE__=1
  define: __SSE2__=1
  define: __FXSR__=1
  define: __SSE_MATH__=1
  define: __SSE2_MATH__=1
  define: __MMX_WITH_SSE__=1
  define: __SEG_FS=1
  define: __SEG_GS=1
  define: __SEH__=1
  define: __stdcall=__attribute__((__stdcall__))
  define: __fastcall=__attribute__((__fastcall__))
  define: __thiscall=__attribute__((__thiscall__))
  define: __cdecl=__attribute__((__cdecl__))
  define: __GXX_MERGED_TYPEINFO_NAMES=0
  define: __GXX_TYPEINFO_EQUALITY_INLINE=0
  define: __MSVCRT__=1
  define: __MINGW32__=1
  define: _WIN32=1
  define: __WIN32=1
  define: __WIN32__=1
  define: __WINNT=1
  define: __WINNT__=1
  define: _INTEGRAL_MAX_BITS=64
  define: __MINGW64__=1
  define: __WIN64=1
  define: __WIN64__=1
  define: _WIN64=1
  define: __declspec(x)=__attribute__((x))
  define: __DECIMAL_BID_FORMAT__=1
  define: _REENTRANT=1
  define: NOMINMAX=1
  define: _USE_MATH_DEFINES
  other: --g++
  other: --gnu_version=100200
  stdver: c++17
  intelliSenseMode: gcc-x64
Queueing IntelliSense update for files in translation unit of: C:\MSYS64\HOME\FLAMEWING\TEMP\TEST.CC
cpptools/finishUpdateSquiggles
Error squiggle count: 2
Update IntelliSense time (sec): 1.739
  Processing folder (recursive): C:/MSYS64/MINGW64/LIB/GCC/X86_64-W64-MINGW32/10.2.0/INCLUDE-FIXED/
  Processing folder (recursive): C:/MSYS64/MINGW64/X86_64-W64-MINGW32/INCLUDE/
  Processing folder (recursive): C:/MSYS64/HOME/FLAMEWING/TEMP/
  Processing folder (recursive): C:/MSYS64/MINGW64/LIB/GLIB-2.0/INCLUDE/
  Processing folder (recursive): C:/MSYS64/MINGW64/LIB/GTK-2.0/INCLUDE/
  Discovering files: 27644 file(s) processed
  0 file(s) removed from database
Done discovering files.
Parsing remaining files...
  Parsing: 0 files(s) processed
Done parsing remaining files.
```
</details>

**Comment by sean-mcmanus** [comment]
Thanks for reporting this. I've filed a bug on VS for our shared parser: https://developercommunity.visualstudio.com/content/problem/1207998/cc-intellisense-gives-errors-regarding-size-t-and.html .

The issue doesn't repro on Linux/Mac or with cl.exe, only gcc mode on Windows, i.e. seems to be some issue with size_t different on Windows.

**Comment by michelleangela** [comment]
Fixed with https://github.com/microsoft/vscode-cpptools/releases/tag/1.3.0-insiders4

**Comment by michelleangela** [comment]
Fix is released in the 1.3.0 release version https://github.com/microsoft/vscode-cpptools/releases/tag/1.3.0
